### PR TITLE
Updated go sdk version

### PR DESCRIPTION
**Pull request by dginty4**

### CODE REVIEW

File: genesyscloud/resource_genesyscloud_routing_email_domain.go [hunk 1]
```diff
@@ -12,24 +12,28 @@ import (
...
 	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
-	"github.com/mypurecloud/platform-client-sdk-go/v80/platformclientv2"
+	"github.com/mypurecloud/platform-client-sdk-go/v89/platformclientv2"
 )
 
 func getAllRoutingEmailDomains(_ context.Context, clientConfig *platformclientv2.Configuration) (ResourceIDMetaMap, diag.Diagnostics) {
 	resources := make(ResourceIDMetaMap)
 	routingAPI := platformclientv2.NewRoutingApiWithConfig(clientConfig)
 
-	domains, _, getErr := routingAPI.GetRoutingEmailDomains(false)
```
Comment (carnellj-genesys) [inline genesyscloud/resource_genesyscloud_routing_email_domain.go:22]: Why was this changed?

File: genesyscloud/resource_genesyscloud_routing_email_route.go [hunk 2]
```diff
@@ -36,35 +36,38 @@ func getAllRoutingEmailRoutes(_ context.Context, clientConfig *platformclientv2.
...
 
-	for _, domain := range *domains.Entities {
-		for pageNum := 1; ; pageNum++ {
-			const pageSize = 100
-			routes, resp, getErr := routingAPI.GetRoutingEmailDomainRoutes(*domain.Id, pageSize, pageNum, "")
-			if getErr != nil {
-				if isStatus404(resp) {
-					// Domain not found
-					break
+		for _, domain := range *domains.Entities {
```
Comment (carnellj-genesys) [inline genesyscloud/resource_genesyscloud_routing_email_route.go:50]: It looks like this code is the same code just formatted differently.  Make sure you run go fmt on your files if you have not done so

File: genesyscloud/resource_genesyscloud_routing_email_domain.go (hunk 1 above)
Comment (dginty4) [reply genesyscloud/resource_genesyscloud_routing_email_domain.go:22]: Some of the methods had changed slightly when I updated the sdk version so I needed to change them a bit to work again

**Review by carnellj-genesys — APPROVED** [review]
